packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -10,15 +10,20 @@ import 'package:meta/meta.dart';
 
 import '../application_package.dart';
 import '../base/common.dart';
+import '../base/context.dart';
 import '../build_info.dart';
 import '../dart/package_map.dart';
 import '../dart/pub.dart';
 import '../device.dart';
+import '../doctor.dart';
 import '../flx.dart' as flx;
 import '../globals.dart';
 import '../usage.dart';
 import 'flutter_command_runner.dart';
 
+DeviceManager get deviceManager => context[DeviceManager];
```

Comment:
These getters are copied several places; should they live in the defining libraries (`devices.dart`; `doctor.dart`). It would mean less code duplication.